peer.go

```diff
@@ -92,6 +94,15 @@ func (id ID) ExtractPublicKey() (ic.PubKey, error) {
 	return pk, nil
 }
 
+// Validate check if ID is empty or not
+func (id ID) Validate() error {
+	if id == ID("") {
+		return errors.New("Empty peer ID")
```

Comment:
This should probably:

1. Be a global `ErrEmptyPeerID`.
2. Be "empty peer ID" (lower case, see: https://github.com/golang/go/wiki/CodeReviewComments#error-strings)